### PR TITLE
refactor remove-worst-server.js

### PR DESCRIPTION
**Pull request by Nytelife26**
This refactor makes use of newer syntax, tidies things up a little, makes the use of `var`/`let`/`const` follow guidelines better, and is more performant as far as I know. Let me know if you want anything changed.

### CODE REVIEW

File: remove-worst-server.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
-            worstServerName = serverName;
-            worstServerRam = ram;
-        }
-    });
-    if (worstServerName == null) {
-        ns.tprint("Nothing to delete - all " + purchasedServers.length + " servers have the maximum " + worstServerRam + " GB of RAM");
-        return;
-    }
+    const purchasedServers = await getNsDataThroughFile(ns, 'ns.getPurchasedServers()', '/Temp/purchased-servers.txt');
+    if (!purchasedServers)
```
Comment (alainbryden) [inline remove-worst-server.js:9]: Is an empty array falsey? I didn't think it was.

Comment (Nytelife26) [reply remove-worst-server.js:9]: For some reason I thought it was, but you're right. My bad.

**Comment by Nytelife26** [comment]
I've observed a strange pattern, actually. `ns.getPurchasedServers()` seems to return servers in order of their RAM - can anyone verify? If this is the case, we can simplify the script even more.

```
[14/04/2022 03:46:19] /Temp/terminal-command.js: [["daemon-16",16384],["daemon-17",16384],["daemon-19",16384],["daemon-20",16384],["daemon-21",16384],["daemon-22",32768],["daemon-23",32768],["daemon",32768],["daemon-0",32768],["daemon-1",32768],["daemon-2",32768],["daemon-3",32768],["daemon-4",32768],["daemon-5",32768],["daemon-6",32768],["daemon-7",32768],["daemon-8",32768],["daemon-9",65536],["daemon-10",65536],["daemon-11",65536],["daemon-12",65536],["daemon-13",65536],["daemon-14",131072],["daemon-15",131072],["daemon-18",262144]]
```

**Comment by alainbryden** [comment]
> I've observed a strange pattern, actually. `ns.getPurchasedServers()` seems to return servers in order of their RAM - can anyone verify? If this is the case, we can simplify the script even more.
> 
> ```
> [14/04/2022 03:46:19] /Temp/terminal-command.js: [["daemon-16",16384],["daemon-17",16384],["daemon-19",16384],["daemon-20",16384],["daemon-21",16384],["daemon-22",32768],["daemon-23",32768],["daemon",32768],["daemon-0",32768],["daemon-1",32768],["daemon-2",32768],["daemon-3",32768],["daemon-4",32768],["daemon-5",32768],["daemon-6",32768],["daemon-7",32768],["daemon-8",32768],["daemon-9",65536],["daemon-10",65536],["daemon-11",65536],["daemon-12",65536],["daemon-13",65536],["daemon-14",131072],["daemon-15",131072],["daemon-18",262144]]
> ```

It's actually in purchase order, I just happen to write scripts that refuse to purchase a server smaller than our current smallest purchased server. Can't rely on that always being the case though.

**Review by alainbryden — APPROVED** [review]

**Pull request merged**